pyproject.toml

```diff
@@ -1,79 +1,61 @@
 [build-system]
-# Uncomment this line, the `meson-python` requires line, and the [project] and
-# [project.urls] tables below in order to build with Meson by default
-#build-backend = "mesonpy"
+build-backend = "mesonpy"
 requires = [
-    # setuptools, wheel and Cython are needed for the setup.py based build
-    "setuptools==59.2.0",
-    # `wheel` is needed for non-isolated builds, given that `meson-python`
-    # doesn't list it as a runtime requirement (at least in 0.11.0) - it's
-    # likely to be removed as a dependency in meson-python 0.12.0.
-    "wheel==0.38.1",
-    "Cython>=0.29.34,<3.0",
-#    "meson-python>=0.10.0",
+    "Cython>=3.0",
```

Comment:
This isn't desirable nor necessary. Things seem to build fine with Cython 0.29.35 locally. It's not desirable because this may force distros to upgrade to Cython 3.0 wholesale. That almost certainly won't work yet in a number of packages.

Also, we need upper bounds for dependencies here. I'll push a fix.

Reply:
The pin in `pyproject.toml.setuppy` is `"Cython>=0.29.34,<3.1",` that does look right.

Reply:
It needs to be changed in the main branch also.